### PR TITLE
Use mamba-org/setup-micromamba GitHub action, support Python3.11

### LINKED ISSUE #1389: ⚠️ Nightly upstream-dev CI failed ⚠️
[Workflow Run URL](https://github.com/Ouranosinc/xclim/actions/runs/5216820068)
<details><summary>Python 3.10 Test Summary</summary>

```
tests/test_ffdi.py::TestFFDI::test_ffdi_indicators[xlim-True]: KeyError: 'wsgsmax'
tests/test_ffdi.py::TestFFDI::test_ffdi_indicators[xlim-False]: KeyError: 'wsgsmax'
tests/test_ffdi.py::TestFFDI::test_ffdi_indicators[discrete-True]: KeyError: 'wsgsmax'
tests/test_ffdi.py::TestFFDI::test_ffdi_indicators[discrete-False]: KeyError: 'wsgsmax'
```

</details>

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1389 
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Updates the CI to use the new suggested action for creating Anaconda environments (`mamba-org/setup-micromamba`)
* Classifies `xclim` as officially supporting Python3.11
* Updates the CI testing ensemble to include a Python3.11 build.

### Does this PR introduce a breaking change?

No.

### Other information:

https://github.com/mamba-org/provision-with-micromamba#migration-to-setup-micromamba

### CODE REVIEW

**Review by aulemahal — APPROVED** [review]
Nice.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ on:
       - submitted
 
 env:
-  XCLIM_TESTDATA_BRANCH: v2023.5.4
+  XCLIM_TESTDATA_BRANCH: v2023.6.8
```
Comment (coxipi) [inline .github/workflows/main.yml:24]: Maybe this part too concerns the PR changing wsgsmax -> sfcWindmax ?

Comment (Zeitsperre) [reply .github/workflows/main.yml:24]: Yep, we have checks to ensure that both the `main` and latest tag of testing data are in sync. This was related to #1358.

File: tests/test_ffdi.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def test_ffdi_indicators(self, open_dataset, init_kbdi, limiting_func):
         assert df.shape == test_data["pr"].shape
 
         ffdi = atmos.mcarthur_forest_fire_danger_index(
-            df, test_data["tasmax"], test_data["hurs"], test_data["wsgsmax"]
+            df, test_data["tasmax"], test_data["hurs"], test_data["sfcWindmax"]
         )
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply tests/test_ffdi.py:180]: This was causing failures on main, so I figured it made more sense to update it here.

**Pull request merged**